android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/SignInViewModel.kt

```diff
@@ -0,0 +1,157 @@
+package com.google.samples.quickstart.canonical
+
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import android.widget.Toast
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.GoogleAuthProvider
+
+
+class SignInViewModel : ViewModel() {
+    data class CurFirebaseUser(
+        var firebaseUser: MutableLiveData<FirebaseUser?> = MutableLiveData(FirebaseAuth.getInstance().currentUser),
+        var isLogin: MutableLiveData<Boolean> = MutableLiveData(firebaseUser.value?.let { true } ?: run{ false })
+    )
+
+    private lateinit var googleSignInClient: GoogleSignInClient
+    private lateinit var context: Context
+    private lateinit var activity: MainActivity
+    private var authStateListenerForSignOut : FirebaseAuth.AuthStateListener? = null
+    private lateinit var curFirebaseUser : MutableLiveData<CurFirebaseUser>
+
+
+    private fun setResources(activityContext: Context, activityMain: MainActivity) {
+        context = activityContext
+        activity = activityMain
+    }
+
+    private fun googleSignInInit() {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(context.getString(R.string.default_web_client_id))
+            .requestEmail()
+            .build()
+
+        googleSignInClient = GoogleSignIn.getClient(activity, gso)
+        Log.d(SIGN_IN_VM_TAG, "googleSignInClientInit")
+    }
```

Comment:
```suggestion
    private fun getGoogleSignInClient() : GoogleSignInClient {
        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
            .requestIdToken(context.getString(R.string.default_web_client_id))
            .requestEmail()
            .build()
            
        Log.d(SIGN_IN_VM_TAG, "googleSignInClientInit")
        return GoogleSignIn.getClient(activity, gso)
    }
```

Seems this can just return a GoogleSignInClient. And the googleSignInClient field on top is not necessary.